simcon_project/users/views/create_assignment.py

```diff
@@ -0,0 +1,153 @@
+from django.shortcuts import render
+from users.models import SubjectLabel, Assignment, Student, Researcher
+from conversation_templates.models import ConversationTemplate
+from django.views.generic import TemplateView
+from django.core import serializers
+from django.http import HttpResponse
+from django.core.mail import send_mail
+from apscheduler.schedulers.background import BackgroundScheduler
+from tzlocal import get_localzone
+from django.contrib.auth.decorators import user_passes_test
+from users.views.researcher_home import is_researcher
+import json
+import datetime
+
+
+class CreateAssignmentView(TemplateView):
+    def get(self, request):
+        curResearcher=request.user
+
+        stud=Student.objects.all().filter(added_by=curResearcher).values('email', 'first_name','last_name','is_active')
+        label=SubjectLabel.objects.all().filter(researcher=curResearcher)
+        template=ConversationTemplate.objects.all().filter(researcher=curResearcher)
+
+        student=json.dumps(list(stud))
+        subLabel = serializers.serialize("json",label)
+        template = serializers.serialize("json",template)
+
+        return render(request, 'create_assignment.html', {
+            'student': student,
+            'subLabel': subLabel,
+            'template': template,
+        })
+
+#Transfer string type list to list type
+def decode(str):
+    if str[0]!='[':
+        print("Decode failed. The data is not a list type.")
+        return
+    temp=str[1:-1]
+    temp=temp.split(',')
+    listTmp=[]
+    for t in temp:
+        t=t[1:-1]
+        listTmp.append(t)
+    return listTmp
+
+def sendMail(subject, msg, recipient, researcher):
+    send_mail(subject,msg,researcher,recipient,fail_silently=False)
```

Comment:
I was having some issues getting emails to send but was able to fix by changing researcher to our email name,                       'simulated.conversation@mail.com'. I think we're switching over to gmail soon but this makes the email sending testable.